htdocs/modules/system/admin/findusers/main.php

```diff
@@ -266,8 +275,8 @@
 		}
 	}
 	if (!empty($user_url)) {
-		$url = formatURL(trim($user_url));
-		$criteria->add(new icms_db_criteria_Item('url', $url . '%', 'LIKE'));
+		$url = filter_var($user_url, FILTER_SANITIZE_URL);
+		$criteria->add(new icms_db_criteria_Item('url', '%' . $url . '%', 'LIKE'));
```

Comment:
I'm not 100% sure here but I think this is better then stripping special characters (it should escape them):
```suggestion
		$criteria->add(new icms_db_criteria_Item('url', '%' . str_replace(['%', '_'], ['\%', '\_'], $user_url) . '%', 'LIKE'));
```

Reply:
Escaping them would make them safe to add to the database - what are the risks when the link is rendered and someone clicks on it?

Reply:
For outputing, I think logic is in getVar method, if correct mode is used it shouldn't be a problem. But in any case is much less risky then the issue with DB. It would be much harder to exploit that.

Reply:
% and _ are valid characters in a URL and there are other characters that should be removed or encoded/escaped. 
Maybe this would be a better filter option? https://www.php.net/manual/en/filter.constants.php#constant.filter-sanitize-encoded

Reply:
Using formatURL() is going to result in the same issue that started this - the user input is going to be converted to something with 'https://' prefixed at the beginning. What we need is an SQL-safe value from user input to query the database for a URL that contains the input, and it should only be for characters that are allowed in a valid URL

Reply:
Yeah... somehow I overlooked that... and AI from small context window not allways correctly returns response... :|

Probably this should be:
```suggestion
$criteria->add(
   new icms_db_criteria_Item(
      'url', 
      '%' . str_replace(['%', '_'], ['\%', '\_'], icms_core_DataFilter::addSlashes(trim($user_url))) . '%',
      'LIKE'
   )
);
```

Also, I noticed others criteria incorrectly escapes special LIKE characters

Reply:
icms_core_DataFilter::checkVar() is already filtering the user input at the beginning of modules/system/admin/findusers/main.php and the default for any variable not specified in filter_post() or filter_get() is to treat as a string input. The default filtering for that is 

`return` htmlspecialchars(strip_tags($data), ENT_NOQUOTES);

Do we need more?

Reply:
icms_core_dataFilter doesn't have anything that would escape correctly data for LIKE queries, so I'm thinking that extra escaping is needed. 

See this little test -> https://onlinephp.io/c/9ee76947-3320-4f64-94e1-63b110f5c054 

But still I'm not sure if there really is a big problem

Reply:
> % and _ are magic symbols in MySQL LIKEs, so what here happening is these symbols are escaping.

I wasn't paying attention to that - it makes sense to escape those characters. I performed a test and searching for '%' returns all users with websites, not just ones with that character.

I still feel we should sanitize the input using FILTER_SANITIZE_URL - why let them search for something that should not be there?

Reply:
Question (still learning) - would switching to prepared statements also be effective in escaping the input prior to executing the query?

Reply:
Switching to prepared statements would also address the overall security issue. However, if I recall correctly, it took me over a month to implement that on the TNG branch. It would be great to backport that work - it would save a lot of time.

About another your question - I'm unsure. Usually if such questions arise I'm writing tests. But how do that for such code, I'm really unsure.